### PR TITLE
add update href in DocumentBuildContext.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @vicancy @hellosnow @qinezh @superyyrrzz

### CODE REVIEW

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 1]
```diff
@@ -240,6 +240,62 @@ public void SetFilePath(string key, string filePath)
...
+            }
+            string filePath;
+            if (!FileMap.TryGetValue(rp.UrlDecode(), out filePath))
+            {
+                return href;
+            }
+            string modifiedPath;
+            if (fromFile == null)
+            {
+                modifiedPath = ((RelativePath)filePath).UrlEncode();
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: modifiedPath => modifiedHref?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
ut failed

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**